Focused-Launch: Update <Title/>, <SubTitle/> & <p/> to improve a11y

StefanNieuwenhuis opened the pull request:
The focused launch modal has to be semantically correct to enable all users to enjoy its features, hence we added tagName props to the aforementioned elements to improve semantics. We also converted `<p/>` to `<SubTitle/>` where necessary to improve consistency.

#### Changes proposed in this Pull Request

* Add `tagName` props to `<Title/>`, `<SubTitle/>` & `<p/>`;
* Convert `<p/>` to `<SubTitle/>`.

#### Testing instructions

* Check out this branch on your machine and run `yarn && yarn start`;
* Navigate to the focused launch flow;
* Click on "View all plans";
* use `devTools` to inspect the "Select a plan" heading and confirm that's a `<h2/>`.
* use `devTools` to inspect the "There's no risk, you can cancel for a full refund within 30 days." subtitle and confirm that's a `<h3/>`.

Fixes #47497

StefanNieuwenhuis committed:
Focused-Launch: Update <Title/>, <SubTitle/> & <p/> to improve a11y

The focused launch modal has to be semantically correct to enable all users to enjoy its features, hence we added tagName props to aforementioned elements to improve semantics. We also converted <p/> to <SubTitle/> where necessary to improve consistency.

ciampo commented on packages/launch/src/focused-launch/summary/index.tsx:
```diff
@@ -130,8 +130,10 @@ const DomainStep: React.FunctionComponent< DomainStepProps > = ( {
 							</Tooltip>
-							{ /* @TODO: should use <Subtitle/> once https://github.com/Automattic/wp-calypso/issues/47418 is solved */ }
-							<p className="focused-launch-summary__mobile-commentary focused-launch-summary__mobile-only">
+							<SubTitle
+								tagName="p"
+								className="focused-launch-summary__mobile-commentary focused-launch-summary__mobile-only"
+							>
```
This `<p>` tag actually didn't need to be converted to `<Subtitle />` — sorry, my bad!
Please revert the change (but still delete the TODO comment)

StefanNieuwenhuis replied:
I just reverted the changes and left the comment out.

ciampo commented on packages/launch/src/focused-launch/summary/index.tsx:
```diff
@@ -518,12 +520,12 @@ const Summary: React.FunctionComponent = () => {
 						: __( "You're almost there", __i18n_text_domain__ ) }
 				</Title>
-				<p className="focused-launch-summary__caption">
+				<SubTitle tagName="p" className="focused-launch-summary__caption">
```
Could also remove the `font-size` and `color` CSS rules from the `.focused-launch-summary__caption` ? We don't need them anymore, now that we're using `<Subtitle />`

StefanNieuwenhuis replied:
Makes total sense and I removed them!

ciampo reviewed (commented):
Thank you for working on these changes! I just left a couple of comments, but overall it looks good!

StefanNieuwenhuis committed:
Focused-Launch: Revert changes to <p/> and remove font-size & color for summary caption

The changes for the focused-launch-summary__mobile-commentary were not necessary and reverted.
Removed font-size & color from `.focused-launch-summary__caption` since we do not need them anymore because it is included in the `<SubTitle/>` base class.

StefanNieuwenhuis commented:
> Thank you for working on these changes! I just left a couple of comments, but overall it looks good!

Thanks for reviewing!

ciampo reviewed (approved):
Thank you for reacting quickly to the feedback!

LGTM
